File: backend/pkg/testdata/gfmt/max-subarray/solutions/wa.go
package main

import (
	"bufio"
	"fmt"
	"os"
	"strconv"
)

func main() {
	scanner := bufio.NewScanner(os.Stdin)
	scanner.Split(bufio.ScanWords)
	
	if !scanner.Scan() {
		return
	}
	n, _ := strconv.Atoi(scanner.Text())
	
	maxSoFar := int64(0)
	currMax := int64(0)
	
	for i := 0; i < n; i++ {
		if !scanner.Scan() {
			break
		}
		val, _ := strconv.ParseInt(scanner.Text(), 10, 64)
		if currMax + val > 0 {
			currMax = currMax + val
		} else {
			currMax = 0
		}
		if currMax > maxSoFar {
			maxSoFar = currMax
		}
	}
	fmt.Println(maxSoFar)
}

File: backend/pkg/testdata/gfmt/max-subarray/solutions/wa.py
import sys

def main():
    input_data = sys.stdin.read().split()
    if not input_data:
        return
    n = int(input_data[0])
    a = [int(x) for x in input_data[1:n+1]]
    
    max_so_far = 0
    curr_max = 0
    for i in range(n):
        curr_max = max(0, curr_max + a[i])
        max_so_far = max(max_so_far, curr_max)
    print(max_so_far)

if __name__ == '__main__':
    main()
